File: bakery/models/purchaseitem.py
from django.db import models

class PurchaseItem(models.Model):
    purchase_order = models.ForeignKey('PurchaseOrder', on_delete=models.CASCADE, related_name='items')
    ingredient = models.ForeignKey('Ingredient', on_delete=models.CASCADE)
    quantity_ordered = models.DecimalField(max_digits=10, decimal_places=2)
    unit_cost_at_purchase = models.DecimalField(max_digits=10, decimal_places=2)

    def __str__(self):
        return getattr(self, 'name', str(self.id))

File: frontend/src/models/PurchaseItem.ts
import type { PurchaseOrder } from "./PurchaseOrder";
import type { Ingredient } from "./Ingredient";

export interface PurchaseItem {
    id?: number;
    purchase_order: PurchaseOrder;
    ingredient: Ingredient;
    quantity_ordered: number;
    unit_cost_at_purchase: number;
}